metaharvest.sh

```diff
@@ -15,154 +15,245 @@ source "${xpath}"/x.funx.sh
 
 # --- Help message -------------------------------------------------------------
 
-read -d '' _helpmsg_metaharvest << EOM || true
+read -d '' _help_metaharvest << EOM || true
 Utility script to fetch metadata from both GEO and ENA referring to a given
-project.
-
-The issue this script tries to solve is that metadata from GEO have the "useful"
-indications, such as sample name, sample type (e.g., case or control), etc...
-which ENA's do not (always) have. This script downloads the ENA metadata and
-cross references it with GEO metadata to obtain a large metadata matrix for
-better usage later.
-
-Different from other x.FASTQ scripts, this one does not run the job in the
-background. It also emits found files to stdout, while other messages are sent
-to stderr.
+project. The issue this script tries to solve is that metadata from GEO have the
+"useful" indications, such as sample name, sample type (e.g., case or control),
+etc., which ENA's do not (always) have. This script downloads the ENA metadata
+and cross references it with GEO metadata to obtain a large metadata matrix for
+better later usage.
 
 Usage:
   metaharvest [-h | --help] [-v | --version]
-  metaharvest -d | --download ENA
-  metaharvest -m | --metadata ENA
+  metaharvest [-x=ENTRY | --extra=ENTRY] [-e | --ena] [-g | --geo] ID
 
 Positional options:
-  -h | --help       Shows this message and exits.
-  -v | --version    Shows this script's version and exits.
-  -d | --download   Fetches the FTP links to download the FASTQ files of an
-                    entire ENA accession.
-  -m | --metadata   Downloads the cross-referenced metadata from GEO and ENA
-                    as one large metadata matrix.
-  ENA               With -d or -m, the ENA accession number for the project to
-                    download, e.g., "PRJNA141411"
+  -h | --help       Shows this help.
+  -v | --version    Shows script's version.
+  -e | --ena        Downloads metadata from ENA database and prints them as a
+                    CSV-formatted text to stdout.
+  -g | --geo        Downloads metadata from GEO database and prints them as a
+                    CSV-formatted text to stdout.
+  -x=ENTRY |        Adds to metadata CSV a trailing extra column for subsequent
```

Comment:
No, if you mean a column with different values specific to each sample. You can only specify a constant value to be included in each row as a default (including of course the null `-x=""`). Entering sample-specific values from the command line might be useful, but it takes a bit more time to implement and is not strictly necessary for me at the moment.

Reply:
No, I mean if I want to add a column with a specific **name** plus a specific value. Something like:
`-x="my_col=my_value"`
But it's not super necessary right now.

Reply:
ah, ok! this is relatively simple. Done in the new commit 0297dcf49a07956dfd7528d092be080a745c4581.
(but I preferred using the colon as a separator, to avoid double =)

Reply:
Great! Bash string extraction is black magic, but great!